ASP.NET minimal hosting model doc migration.

LayersOfAbstraction opened the pull request:
<!---
Pull Requests for Quickstart Guides can still be submitted here, but most other documentation content is no longer hosted on GitHub and therefore no longer open-sourced. If you are an Auth0 employee trying to make a change, please [submit a ticket](https://auth0team.atlassian.net/servicedesk/customer/portal/9). Thank you!
--->
Hi as requested in [the #79 issue outside of this repo](https://github.com/auth0-samples/auth0-aspnetcore-mvc-samples/issues/79) I had edited the files for the ASP.NET quickstart you mentioned. Here are the doc enhancements. 

FYI:
- All I did was swap in the markdown files for the Startup code for how to use the minimal hosting model logic in program.cs. 
- But I couldn't change the [code here](https://github.com/auth0/docs/blob/master/articles/quickstart/webapp/aspnet-core/interactive.md#:~:text=Install%20and%20Configure%20the%20SDK%20%7B%7B%7B%20data%2Daction%3Dcode%20data%2Dcode%3D%22Startup.cs%239%3A13%22%20%7D%7D%7D) to reflect the desired lines of Program.cs as I cannot verify what they are and do not have access to the framework to run and test the docs locally.
- I cannot submit this to Wordy unless I am reimbursed for submitting this to one of their editors. I imagine that's not even an issue.

Moving on to the WebAPI sample and doc refactoring now, thanks.

LayersOfAbstraction committed:
Replace Startup.cs logic with Program.cs minimal hosting model logic under heading #Install and configure the SDK#.

LayersOfAbstraction committed:
Replace Startup.cs logic with Program.cs minimal hosting model logic under heading #Configure the middleware#.

LayersOfAbstraction committed:
Add file.

LayersOfAbstraction committed:
Merge branch 'master' of https://github.com/LayersOfAbstraction/docs

LayersOfAbstraction committed:
Erase repeated code.

LayersOfAbstraction committed:
Merge branch 'auth0:master' into Minimal_Hosting_Model_Doc_Migration

frederikprijck commented on articles/quickstart/webapp/aspnet-core/01-login.md:
```diff
@@ -41,44 +41,38 @@ Install-Package Auth0.AspNetCore.Authentication
-public void ConfigureServices(IServiceCollection services)
+var builder = WebApplication.CreateBuilder(args);
+// Cookie configuration for HTTP to support cookies with SameSite=None
+services.ConfigureSameSiteNoneCookies();
```
```suggestion
builder.Services.ConfigureSameSiteNoneCookies();
```

frederikprijck commented on articles/quickstart/webapp/aspnet-core/01-login.md:
```diff
@@ -41,44 +41,38 @@ Install-Package Auth0.AspNetCore.Authentication
+    options.Domain = builder.Configuration["Auth0:Domain"];
+    options.ClientId = builder.Configuration["Auth0:ClientId"];
+});
+services.AddControllersWithViews();
```
```suggestion
builder.Services.AddControllersWithViews();
```

frederikprijck commented on articles/quickstart/webapp/aspnet-core/01-login.md:
```diff
@@ -41,44 +41,38 @@ Install-Package Auth0.AspNetCore.Authentication
 :::
 
-Make sure you have enabled authentication and authorization in your `Startup.Configure` method:
+Make sure you have enabled authentication and authorization below your `var app = builder.Build();` middleware pipeline setup statement in `Program.cs`.
```
```suggestion
Make sure you have enabled authentication and authorization in your Program.cs file:
```

frederikprijck commented on articles/quickstart/webapp/aspnet-core/files/Program.md:
````diff
@@ -0,0 +1,16 @@
+
+app.UseAuthentication();
+app.UseAuthorization();
+```
````
I assume we want to remove https://github.com/auth0/docs/blob/master/articles/quickstart/webapp/aspnet-core/files/startup.md ?

frederikprijck replied:
We will also need to remove it from the files array in the frontmatter on interactive.md, and ensure to add the `files/program` file instead.

LayersOfAbstraction replied:
Done. Yes it would be wise to get rid of unused files. Now you just have to delete or comment on this line in interactive.md. 

`## Install and Configure the SDK {{{ data-action=code data-code="Startup.cs#9:13" }}}`

As you know I do not have access to the tool needed to test the content locally or have the knowledge to act on that content.

frederikprijck replied:
No worries, I am happy to try this once reviewed and make any adjustments we missed.

frederikprijck commented on articles/quickstart/webapp/aspnet-core/interactive.md:
```diff
@@ -39,12 +39,11 @@ Install-Package Auth0.AspNetCore.Authentication
 ### Configure the middleware
 
-To enable authentication in your ASP.NET Core application, use the middleware provided by the SDK. Go to the `ConfigureServices` method of your `Startup` class and call `services.AddAuth0WebAppAuthentication()` to register the SDK's middleware.
+To enable authentication in your ASP.NET Core application, use the middleware provided by the SDK. Under the `var builder = WebApplication.CreateBuilder(args);` method of your `Program` class call `builder.Services.AddAuth0WebAppAuthentication()` to register the SDK's middleware.
```
```suggestion
To enable authentication in your ASP.NET Core application, use the middleware provided by the SDK. Go to the `Program.cs` file and call `builder.Services.AddAuth0WebAppAuthentication()` to register the SDK's middleware.
```

frederikprijck commented on articles/quickstart/webapp/aspnet-core/interactive.md:
```diff
@@ -39,12 +39,11 @@ Install-Package Auth0.AspNetCore.Authentication
 
-Make sure you have enabled authentication and authorization in your `Startup.Configure` method.
-
+Make sure you have enabled authentication and authorization below your `var app = builder.Build();` middleware pipeline setup statement in `Program.cs`.
```
```suggestion
Make sure you have enabled authentication and authorization in your `Program.cs` file.
```

frederikprijck reviewed (changes requested):
Thanks for the PR!

LayersOfAbstraction committed:
Update articles/quickstart/webapp/aspnet-core/01-login.md

Co-authored-by: Frederik Prijck <frederik.prijck@gmail.com>

LayersOfAbstraction committed:
Update articles/quickstart/webapp/aspnet-core/01-login.md

Co-authored-by: Frederik Prijck <frederik.prijck@gmail.com>

LayersOfAbstraction committed:
Update articles/quickstart/webapp/aspnet-core/01-login.md

I would personally have left this as it is to make the doc more inclusive but in this case, I suppose the people playing with the quickstart should have already done a bit of .NET 6.

Co-authored-by: Frederik Prijck <frederik.prijck@gmail.com>

LayersOfAbstraction committed:
Update articles/quickstart/webapp/aspnet-core/interactive.md

Co-authored-by: Frederik Prijck <frederik.prijck@gmail.com>

LayersOfAbstraction committed:
Update articles/quickstart/webapp/aspnet-core/interactive.md

Co-authored-by: Frederik Prijck <frederik.prijck@gmail.com>

LayersOfAbstraction committed:
Delete file.

LayersOfAbstraction committed:
Change frontmatter files array value from files/startup to files/program.

LayersOfAbstraction committed:
Merge branch 'auth0:master' into Minimal_Hosting_Model_Doc_Migration

frederikprijck commented on articles/quickstart/webapp/aspnet-core/01-login.md:
```diff
@@ -41,44 +41,38 @@ Install-Package Auth0.AspNetCore.Authentication
 ### Install and configure the SDK
 
-To enable authentication in your ASP.NET Core application, use the middleware provided by the SDK. Go to the `ConfigureServices` method of your `Startup` class and call `services.AddAuth0WebAppAuthentication()` to configure the Auth0 ASP.NET Core SDK.
+To enable authentication in your ASP.NET Core application, use the middleware provided by the SDK. Go to the  `Program` class and between `var builder = WebApplication.CreateBuilder(args);` and `var app = builder.Build();` call `builder.Services.AddAuth0WebAppAuthentication()` to configure the Auth0 ASP.NET Core SDK.
```
```suggestion
To enable authentication in your ASP.NET Core application, use the middleware provided by the SDK. Go to the `Program.cs` file and call `builder.Services.AddAuth0WebAppAuthentication()` to configure the Auth0 ASP.NET Core SDK.
```

LayersOfAbstraction replied:
Thanks. I learned a lot there.

frederikprijck commented on articles/quickstart/webapp/aspnet-core/files/Program.md:
```diff
@@ -0,0 +1,16 @@
+{
+    options.Domain = builder.Configuration["Auth0:Domain"];
+    options.ClientId = builder.Configuration["Auth0:ClientId"];
+});
```
Can we switch the order here to be in line with the 01-login.md example?

frederikprijck commented on articles/quickstart/webapp/aspnet-core/files/Program.md:
```diff
@@ -0,0 +1,16 @@
+    options.ClientId = builder.Configuration["Auth0:ClientId"];
+});
+
+app.UseAuthentication();
```
Can we add what builder and app are here ? Let's create the builde at the top, and the app before using it here.

LayersOfAbstraction replied:
Done. Let me know if I missed anything.

LayersOfAbstraction committed:
Copy suggested changes from 01-login.md.

LayersOfAbstraction committed:
Merge branch 'Minimal_Hosting_Model_Doc_Migration' of https://github.com/LayersOfAbstraction/docs into Minimal_Hosting_Model_Doc_Migration

LayersOfAbstraction committed:
Update articles/quickstart/webapp/aspnet-core/01-login.md

Co-authored-by: Frederik Prijck <frederik.prijck@gmail.com>

frederikprijck committed:
remove Program.md

frederikprijck committed:
Add program.md

frederikprijck reviewed (approved):
Added a commit to rename Program.md to program.md and ensure it contains only the code we had before.

Thanks 🚀

LayersOfAbstraction commented:
Awesome let me know if I missed anything in the 4 other PRs I mentioned in the above issue. Looking forward to merging. Also at the end of all of this if you want to provide some feedback that's fine. Trying to make stuff as manageable as possible but maybe I could have proactively done a CTRL +F for references to `startup.cs` and changed to `program.cs`.
